packages/engineer/feature/gui.dev-server.env.ts

```diff
@@ -0,0 +1,98 @@
+import guiFeature, { mainDashboardEnv } from './gui.feature';
+import { devServerEnv } from './dev-server.feature';
+import VirtualModulesPlugin from 'webpack-virtual-modules';
+import fs from '@file-services/node';
+import type webpack from 'webpack';
+import HtmlWebpackPlugin from 'html-webpack-plugin';
+import { createEntrypoint } from '@wixc3/engine-scripts';
+import { SetMultiMap } from '@wixc3/engine-core';
+import type { IConfigDefinition } from '@wixc3/engine-scripts';
+
+const createDashboardConfig = ({
+    baseConfig,
+    entryPath,
+    basePlugins,
+    virtualModules,
+    title,
+    outputPath,
+}: {
+    baseConfig: webpack.Configuration;
+    entryPath: string;
+    basePlugins: webpack.Plugin[];
+    virtualModules: Record<string, string>;
+    title?: string;
+    outputPath: string;
+}): webpack.Configuration => ({
+    ...baseConfig,
+    entry: {
+        index: entryPath,
+    },
+    target: 'web',
+    plugins: [
+        ...basePlugins,
+        new HtmlWebpackPlugin({
+            filename: `${mainDashboardEnv.env}.html`,
+            chunks: ['index'],
+            title,
+        }),
+        new VirtualModulesPlugin(virtualModules),
+    ],
+    mode: 'development',
+    devtool: 'source-map',
+    output: {
+        ...baseConfig.output,
+        path: outputPath,
+        filename: `[name].web.js`,
+        chunkFilename: `[name].web.js`,
+    },
+});
+
+guiFeature.setup(
+    devServerEnv,
+    (
+        { engineerConfig: { features } },
+        {
+            buildFeature: {
+                engineerWebpackConfigs,
+                devServerConfig: { title, publicConfigsRoute },
+                serverListeningHandlerSlot,
+                application,
+            },
+        }
+    ) => {
+        const baseConfigPath = fs.findClosestFileSync(__dirname, 'webpack.config.js');
+        const baseConfig: webpack.Configuration = typeof baseConfigPath === 'string' ? require(baseConfigPath) : {};
+        const virtualModules: Record<string, string> = {};
+
+        const { plugins: basePlugins = [] } = baseConfig;
+        const entryPath = fs.join(__dirname, 'main-dashboard-web-entry.js');
+        const configurations = new SetMultiMap<string, IConfigDefinition>();
+
+        virtualModules[entryPath] = createEntrypoint({
+            features,
+            childEnvs: [],
+            envName: mainDashboardEnv.env,
+            mode: 'development',
+            publicConfigsRoute,
+            staticBuild: false,
+            configurations,
+        });
+
+        engineerWebpackConfigs.register(
+            createDashboardConfig({
```

Comment:
Yes, again, this is a step forward, originally the dashboard was injected into the target applications compiler as a new entrypoint. I started off with breaking it out into it's own compiler. Moving forward a believe we can take 2 paths
1. Pre build the dashboard - it will give us better performance and it makes more sense than building it everytime the user tries to run his application but (unless I am mistaken) will not allow the user to extend the application
2. Keep on building on the fly, which is the same experience as we currently have, but will leave us the option for extnedability, which I think is what we are aiming for. 

I assume there are other middle ground options, but I haven't dived deep into this problem, I just started with what didn't make sense at all, and left the other decisions for later. @AviVahl